src/scheduler.cpp

```diff
@@ -62,4 +63,7 @@ void Scheduler::shutdown()
 	});
 }
 
-SchedulerTask* createSchedulerTask(uint32_t delay, TaskFunc&& f) { return new SchedulerTask(delay, std::move(f)); }
+SchedulerTask_ptr createSchedulerTask(uint32_t delay, TaskFunc&& f)
+{
+	return SchedulerTask_ptr(new SchedulerTask(delay, std::move(f)));
```

Comment:
The correct way would be to use `std::make_unique` instead of calling `new`
There are many advantages to using make_unique

less allocations, safe in case of exceptions, and it's the modern way of doing things